recipes/gdal/post_3.5.0/conanfile.py

```diff
@@ -314,591 +331,487 @@ def validate(self):
-
-        cmake.definitions["GDAL_USE_ZSTD"] = self.options.with_zstd
-        if self.options.with_zstd:
-            cmake.definitions["GDAL_CONAN_PACKAGE_FOR_ZSTD"] = "zstd"
-            cmake.definitions["TARGET_FOR_ZSTD"] = \
-                    self.dependencies["zstd"].cpp_info.get_property("cmake_target_name")
-        else:
-            cmake.definitions["ZSTD_FOUND"] = False
-
-
-        for k, v in cmake.definitions.items():
-            print(k, " = ", v)
-
-        cmake.configure(build_folder=self._build_subfolder)
-        return cmake
-
-    def build(self):
+            tc.variables["SQLite3_HAS_COLUMN_METADATA"] = self.dependencies["sqlite3"].options.enable_column_metadata
+            tc.variables["SQLite3_HAS_RTREE"] = self.dependencies["sqlite3"].options.enable_rtree
+            tc.variables["SQLite3_HAS_LOAD_EXTENSION"] = not self.dependencies["sqlite3"].options.omit_load_extension
+            tc.variables["SQLite3_HAS_PROGRESS_HANDLER"] = True
+            tc.variables["SQLite3_HAS_MUTEX_ALLOC"] = True
+            tc.preprocessor_definitions["SQLite3_HAS_COLUMN_METADATA"] = 1 if self.dependencies["sqlite3"].options.enable_column_metadata else 0
+            tc.preprocessor_definitions["SQLite3_HAS_RTREE"] = 1 if self.dependencies["sqlite3"].options.enable_rtree else 0
+        # https://github.com/OSGeo/gdal/blob/v3.8.0/cmake/helpers/CheckDependentLibraries.cmake#L419-L450
+        tc.variables["HAVE_JPEGTURBO_DUAL_MODE_8_12"] = (
+                self.options.with_jpeg == "libjpeg-turbo" and
+                bool(self.dependencies["libjpeg-turbo"].options.get_safe("enable12bit"))
+        )
+        # https://github.com/OSGeo/gdal/blob/v3.8.0/port/CMakeLists.txt
+        tc.variables["BLOSC_HAS_BLOSC_CBUFFER_VALIDATE"] = (
+                self.options.with_blosc and
+                Version(self.dependencies["c-blosc"].ref.version) >= "1.21.5"
+        )
+        # https://github.com/OSGeo/gdal/blob/v3.8.0/frmts/hdf5/CMakeLists.txt#L61-L64
+        tc.variables["GDAL_ENABLE_HDF5_GLOBAL_LOCK"] = (
+                self.options.with_hdf5 and
+                bool(self.dependencies["hdf5"].options.get_safe("threadsafe"))
+        )
+        # https://github.com/OSGeo/gdal/blob/v3.8.0/frmts/hdf4/CMakeLists.txt#L28-L46
+        tc.variables["HDF4_HAS_MAXOPENFILES"] = (
+                self.options.with_hdf4 and
+                Version(self.dependencies["hdf4"].ref.version) >= "4.2.5"
+        )
+        # https://github.com/OSGeo/gdal/blob/4bb78aab3ae9ab5433042bc27239d1555cbe272e/cmake/helpers/CheckDependentLibraries.cmake#L301-L318
+        # The detection fails for some reason
+        # Setting it to non-const is compatible with all platforms
+        tc.variables["_ICONV_SECOND_ARGUMENT_IS_NOT_CONST"] = True
+        tc.generate()
+
+        jsonc = self.dependencies["json-c"]
+        jsonc.cpp_info.includedirs.append(os.path.join(jsonc.package_folder, "include", "json-c"))
```

Comment:
What is this trying to acomplish?
As per https://docs.conan.io/2/reference/conanfile/methods/generate.html#dependencies-interface

> All the information is read only. Any attempt to modify dependencies information is an error and can raise at any time, even if it doesn’t raise yet.

Reply:
Oh, that's a bit disappointing to hear. That approach has been helpful in multiple recipes.

Includes against the vendored `json-c` in the recipe don't use a directory prefix and this was one way to fix that.

Maybe that design choice could be reconsidered in Conan to allow tweaking of dependency info for the build only. It's similar to `ConanDeps.set_property()`, imo.

Reply:
I've created the PR https://github.com/conan-io/conan-center-index/pull/22752 to include the extra folder to json-c. Indeed gdal is correct in include that path directly.